projects/step-frontend/src/lib/modules/timeseries/components/table-dashlet/table-dashlet.component.html

```diff
@@ -0,0 +1,104 @@
+<div class="header-container">
+  <span class="uplot title">
+    {{ item.name }}
+  </span>
+  <div class="settings-container">
+    @if (editMode) {
+      <div class="edit-buttons">
+        <button mat-icon-button matTooltip="Shift left" (click)="shiftLeft.emit()">
+          <step-icon name="chevron-left"></step-icon>
+        </button>
+        <button mat-icon-button matTooltip="Shift right" (click)="shiftRight.emit()">
+          <step-icon name="chevron-right"></step-icon>
+        </button>
+        <button mat-icon-button matTooltip="Edit settings" (click)="openSettings()">
+          <step-icon name="edit"></step-icon>
+        </button>
+        <button mat-icon-button matTooltip="Remove chart" (click)="remove.emit()">
+          <step-icon name="x"></step-icon>
+        </button>
+      </div>
+    }
+    @if (!editMode) {
+      <button mat-icon-button matTooltip="Table Settings" [matMenuTriggerFor]="settingsMenu">
+        <step-icon name="settings" />
+      </button>
+      <mat-menu #settingsMenu="matMenu">
+        <button mat-menu-item [matMenuTriggerFor]="columnsMenu">Columns</button>
+      </mat-menu>
+    }
+    <mat-menu #columnsMenu="matMenu">
+      @for (column of columnsDefinition; track column) {
```

Comment:
For the iteration of the complex objects, it's recommended to use one of it's fields as track value, that can be use for entry identification.